build-system/tasks/visual-diff.rb

```diff
@@ -0,0 +1,146 @@
+#!/usr/bin/ruby
+#
+# Copyright 2017 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+# Visual diff generator for AMP webpages using Percy.
+
+
+require 'percy/capybara/anywhere'
+require 'capybara/poltergeist'
+require 'phantomjs'
+require 'json'
+require "net/http"
+
+
+ENV['PERCY_DEBUG'] = '1'  # Enable debugging output.
+DEFAULT_WIDTHS = [375, 411]  # CSS widths: iPhone: 375, Pixel: 411.
+HOST = 'localhost'
+PORT = '8000'
+
+
+# Launches a background AMP webserver for unminified js using gulp.
+#
+# Returns:
+# - Process ID of server process.
+def launchWebServer()
+  webserverCmd = 'gulp serve --host ' + HOST + ' --port ' + PORT
+  webserverUrl = 'http://' + HOST + ':' + PORT
+  @pid = fork do
```

Comment:
This is for readability, to explicitly indicate that we are returning the value  of `pid` at the end of the function.